src/providers/log.provider.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable:no-any
+import {Provider} from '@loopback/context';
+import {ParsedRequest} from '@loopback/core';
+
+export interface LogFn {
+  (req: ParsedRequest, args: any, result: any, startTime?: number[]): void;
+}
+
+export class LogProvider implements Provider<LogFn> {
```

Comment:
I am proposing to rename this class to `LogActionProvider` and the file to `log-action.provider.ts`.